src/lib/interruptible/interruptible.mli

```diff
@@ -0,0 +1,52 @@
+open Core_kernel
+open Async_kernel
+
+(** The type of interruptible computations.
+    [('a, 's) t] represents a computation that produces a value of type ['a],
+    but which may be interrupted by a signal of type ['s].
```

Comment:
might be nice to say what a "signal" is in this context (it's not a Unix signal)